src/main/java/org/asteriskjava/live/internal/AsteriskQueueImpl.java

```diff
@@ -354,27 +354,34 @@ void createNewEntry(AsteriskChannelImpl channel, int reportedPosition, Date date
     {
         AsteriskQueueEntryImpl qe = new AsteriskQueueEntryImpl(server, this, channel, reportedPosition, dateReceived);
 
-        long delay = serviceLevel * 1000L;
-        if (delay > 0)
-        {
-            ServiceLevelTimerTask timerTask = new ServiceLevelTimerTask(qe);
-            timer.schedule(timerTask, delay);
-            synchronized (serviceLevelTimerTasks)
-            {
-                serviceLevelTimerTasks.put(qe, timerTask);
-            }
-        }
-
         synchronized (entries)
         {
-            entries.add(qe); // at the end of the list
+			if (getEntry(channel.getName()) != null)
+			{
+				logger.debug("Ignored duplicate queue entry during population in queue " + name + " for channel "
+					+ channel.getName());
+				return;
+			}
+
+			entries.add(qe); // at the end of the list
 
             // Keep the lock !
             // This will fire PCE on the newly created queue entry
             // but hopefully this one has no listeners yet
             shift();
         }
 
+		long delay = serviceLevel * 1000L;
```

Comment:
What was the reason for moving this code after the queue add?

Reply:
this makes sense to me, if the entry wasn't added then there is no need to schedule a sevicelevel task